src/open-api-helpers.ts

```diff
@@ -0,0 +1,653 @@
+  return Object.keys(shape).reduce(
+    (carry, key) => ({
+      ...carry,
+      [key]: depictSchema({ schema: shape[key], isResponse }),
+    }),
+    {} as Record<string, SchemaObject>
+  );
+};
+
+const depictEffect: DepictHelper<z.ZodEffects<z.ZodTypeAny>> = ({
+  schema,
+  initial,
+  isResponse,
+}) => {
+  const input = depictSchema({ schema: schema._def.schema, isResponse });
+  const effect = schema._def.effect;
+  if (isResponse && effect && effect.type === "transform") {
+    let output = "undefined";
+    try {
+      output = typeof effect.transform(
+        ["integer", "number"].includes(`${input.type}`)
+          ? 0
+          : "string" === input.type
+          ? ""
+          : "boolean" === input.type
+          ? false
+          : "object" === input.type
+          ? {}
+          : "null" === input.type
+          ? null
+          : "array" === input.type
+          ? []
+          : undefined
+      );
+    } catch (e) {
+      /**/
+    }
+    return {
+      ...initial,
+      ...input,
+      ...(["number", "string", "boolean"].includes(output)
+        ? {
+            type: output as "number" | "string" | "boolean",
+          }
+        : {}),
+    };
+  }
+  if (!isResponse && effect && effect.type === "preprocess") {
+    const { type: inputType, ...rest } = input;
+    return {
+      ...initial,
+      ...rest,
+      format: `${rest.format || inputType} (preprocessed)`,
+    };
+  }
+  return { ...initial, ...input };
+};
+
+const depictIOExamples = <T extends IOSchema>(
+  schema: T,
+  isResponse: boolean,
+  omitProps: string[] = []
+): MediaExamples => {
+  const examples = getExamples(schema, isResponse);
+  if (examples.length === 0) {
+    return {};
+  }
+  return {
+    examples: examples.reduce<ExamplesObject>(
+      (carry, example, index) => ({
+        ...carry,
+        [`example${index + 1}`]: <ExampleObject>{
+          value: omit(omitProps, example),
+        },
+      }),
+      {}
+    ),
+  };
+};
+
+const depictIOParamExamples = <T extends IOSchema>(
+  schema: T,
+  isResponse: boolean,
+  param: string
+): MediaExamples => {
+  const examples = getExamples(schema, isResponse);
+  if (examples.length === 0) {
+    return {};
+  }
+  return {
+    examples: examples.reduce<ExamplesObject>(
+      (carry, example, index) =>
+        param in example
+          ? {
+              ...carry,
+              [`example${index + 1}`]: <ExampleObject>{
+                value: example[param],
+              },
+            }
+          : carry,
+      {}
+    ),
+  };
+};
+
+export const depictRequestParams = ({
+  path,
+  method,
+  endpoint,
+}: ReqResDepictHelperCommonProps): ParameterObject[] => {
+  const schema = endpoint.getInputSchema();
+  const shape = extractObjectSchema(schema).shape;
+  const pathParams = getRoutePathParams(path);
+  return Object.keys(shape)
+    .filter((name) => method === "get" || pathParams.includes(name))
```

Comment:
![image](https://user-images.githubusercontent.com/13189514/142779271-9a53ba83-c67c-4557-8d87-0910a971a643.png)

Reply:
![image](https://user-images.githubusercontent.com/13189514/142779467-ac7f041c-914e-4296-b314-cf4487573736.png)

Reply:
fixed in 50bdd07d5d98e8305e456d51612cbf58de0770f4.
Including DELETE requests here too.
Should involve config/inputSources in the future v4